assets/app/view/game/buy_trains.rb

```diff
@@ -129,10 +129,21 @@ def render
       end
 
       def from_depot(depot_trains)
+        trains_from_depot = buyable_trains_in_depot(depot_trains)
+        @corporation.abilities(:train_discount) do |ability|
+          trains_from_depot.concat(buyable_trains_in_depot(depot_trains, ability))
+        end
+        trains_from_depot
+      end
+
+      def buyable_trains_in_depot(depot_trains, ability = nil)
         depot_trains.flat_map do |train|
+          trains_on_interest = ability&.trains || [train.name]
```

Comment:
trains_of_interest

Reply:
this code doesn't make much sense to me, i think if we removed this next statement we could just call the whole function once

Reply:
See picture above.
So is it possible to get two lines for the available 3T with different prices, with just one call to the method?

Reply:
Different solution in later version, so not relevant any longer.